app/models/badge_rule.rb

```diff
@@ -36,9 +42,14 @@ def rule_satisfied?(test_passage)
   private
 
   def user_completed_all_backend_tests?(user)
-    backend_tests = Test.by_category('Backend')
-    completed_tests = user.test_passages.passed.map(&:test_id).last(1)
-    backend_tests.pluck(:id).all? { |test_id| completed_tests.include?(test_id) }
+    backend_tests_ids = Test.by_category('Backend').pluck(:id)
+    # [19, 4]
+    completed_tests = user.test_passages.passed.pluck(:test_id)
+    # [19, 4, 19, 4, 3, 1]
+    filtered_completed_tests = completed_tests.select { |test_id| backend_tests_ids.include?(test_id) }
+    # [19, 4, 19, 4]
+    backend_tests_ids.size == filtered_completed_tests.uniq.size && filtered_completed_tests.tally.values.uniq.one?
+    # [2] == [2] && [19, 4, 19, 4].tally.values = [2, 2], [2, 2].uniq.one? = true
```

Comment:
What happened if we get in filtered_completed_tests this data:
[19, 4, 19, 4, 4, 4, 4]

Reply:
We will get a false response and this will be correct in terms of triggering the rules

Reply:
[2] == [2] && [19, 4, 19, 4, 4, 4, 4].tally.values = [2, 5], [2, 5].uniq.one? = false
false